src/components/VenueInformation.js

```diff
@@ -59,22 +60,59 @@ const VenueInformation = () => {
 
 	Listener({ setFullAddress, setPhoneNumber, setWebsite });
 
+	useEffect(() => {
+		const getData = setTimeout(() => {
```

Comment:
Why does this have to be in a timeout?

Reply:
The timeout here is used as a debounce function to ensure the data is only queried 2 seconds after typing has stopped. Let me know if you've got a better way to do this, totally open to updating this!

Reply:
Have a look at useDebounce from the @wordpress/components package. 

I used it with the same intention to delay searching while typing in the new venue block.

https://github.com/carstingaxion/gatherpress-venue/blob/f88dd091147f4b2876ad21955a1533a9f0879503/src/components/VenuePostsCombobox.js#L57

Reply:
Thanks for the tip, @carstingaxion. Do you know why this might not fire? I tried implementing this according to your example and it didn't seem to work. I'm unsure what I might be doing wrong? I can share code if that helps

Reply:
Yeah, of course, share your code and I'll gonna have a look in 20 minutes.

Reply:
`/**
 * WordPress dependencies.
 */
import { TextControl } from '@wordpress/components';
import { __ } from '@wordpress/i18n';
import { useDispatch, useSelect } from '@wordpress/data';
import { useState, useEffect } from '@wordpress/element';
import { useDebounce } from '@wordpress/compose';

/**
 * Internal dependencies.
 */
import { Broadcaster, Listener } from '../helpers/broadcasting';

/**
 * VenueInformation component for GatherPress.
 *
 * This component allows users to input and update venue information, including full address,
 * phone number, and website. It uses the `TextControl` component from the Gutenberg editor
 * package to provide input fields for each type of information. The entered data is stored
 * in post meta as JSON and updated using the `editPost` method from the editor package.
 *
 * @since 1.0.0
 *
 * @return {JSX.Element} The rendered React component.
 */
const VenueInformation = () => {
	const editPost = useDispatch('core/editor').editPost;
	// eslint-disable-next-line react-hooks/exhaustive-deps
	const updateVenueMeta = (metaData) => {
		const payload = JSON.stringify({
			...venueInformationMetaData,
			...metaData,
		});
		const meta = { gatherpress_venue_information: payload };

		editPost({ meta });
	};

	let venueInformationMetaData = useSelect(
		(select) =>
			select('core/editor').getEditedPostAttribute('meta')
				.gatherpress_venue_information
	);

	if (venueInformationMetaData) {
		venueInformationMetaData = JSON.parse(venueInformationMetaData);
	} else {
		venueInformationMetaData = {};
	}

	const [fullAddress, setFullAddress] = useState(
		venueInformationMetaData.fullAddress ?? ''
	);
	const [phoneNumber, setPhoneNumber] = useState(
		venueInformationMetaData.phoneNumber ?? ''
	);
	const [website, setWebsite] = useState(
		venueInformationMetaData.website ?? ''
	);

	Listener({ setFullAddress, setPhoneNumber, setWebsite });

	const getLatLngData = useDebounce(() => {
		console.log('test');
		let lat = 0;
		let lng = 0;
		fetch(`https://nominatim.openstreetmap.org/search?q=${fullAddress}&format=geojson`).then(response => {
				// Check if the response is successful
				if (!response.ok) {
					throw new Error(
						sprintf(__('Network response was not ok %s', 'gatherpress'), response.statusText)
					);
				}
				// Parse the JSON from the response
				return response.json();
			})
			.then((data) => {
				// Process the data
				if (data.features.length > 0) {
					lat = data.features[0].geometry.coordinates[1];
					lng = data.features[0].geometry.coordinates[0];
				}
				updateVenueMeta({
					latitude: lat,
					longitude: lng,
				});
			})
			.catch((error) => {
				// Handle any errors
				console.error(
					sprintf(__('There was a problem with the fetch operation: %s', 'gatherpress'), error)
				);
			});
	}, 2000);

	return (
		<>
			<TextControl
				label={__('Full Address', 'gatherpress')}
				value={fullAddress}
				onChange={(value) => {
					Broadcaster({ setFullAddress: value });
					updateVenueMeta({ fullAddress: value });
					getLatLngData();
				}}
			/>
			<TextControl
				label={__('Phone Number', 'gatherpress')}
				value={phoneNumber}
				onChange={(value) => {
					Broadcaster({ setPhoneNumber: value });
					updateVenueMeta({ setPhoneNumber: value });
				}}
			/>
			<TextControl
				label={__('Website', 'gatherpress')}
				value={website}
				type="url"
				onChange={(value) => {
					Broadcaster({ setWebsite: value });
					updateVenueMeta({ setWebsite: value });
				}}
			/>
		</>
	);
};

export default VenueInformation;
`

Reply:
```/**
 * WordPress dependencies.
 */
import { TextControl } from '@wordpress/components';
import { __ } from '@wordpress/i18n';
import { useDispatch, useSelect } from '@wordpress/data';
import { useState, useEffect } from '@wordpress/element';
import { useDebounce } from '@wordpress/compose';

/**
 * Internal dependencies.
 */
import { Broadcaster, Listener } from '../helpers/broadcasting';

/**
 * VenueInformation component for GatherPress.
 *
 * This component allows users to input and update venue information, including full address,
 * phone number, and website. It uses the `TextControl` component from the Gutenberg editor
 * package to provide input fields for each type of information. The entered data is stored
 * in post meta as JSON and updated using the `editPost` method from the editor package.
 *
 * @since 1.0.0
 *
 * @return {JSX.Element} The rendered React component.
 */
const VenueInformation = () => {
	const editPost = useDispatch('core/editor').editPost;
	// eslint-disable-next-line react-hooks/exhaustive-deps
	const updateVenueMeta = (metaData) => {
		const payload = JSON.stringify({
			...venueInformationMetaData,
			...metaData,
		});
		const meta = { gatherpress_venue_information: payload };

		editPost({ meta });
	};

	let venueInformationMetaData = useSelect(
		(select) =>
			select('core/editor').getEditedPostAttribute('meta')
				.gatherpress_venue_information
	);

	if (venueInformationMetaData) {
		venueInformationMetaData = JSON.parse(venueInformationMetaData);
	} else {
		venueInformationMetaData = {};
	}

	const [fullAddress, setFullAddress] = useState(
		venueInformationMetaData.fullAddress ?? ''
	);
	const [phoneNumber, setPhoneNumber] = useState(
		venueInformationMetaData.phoneNumber ?? ''
	);
	const [website, setWebsite] = useState(
		venueInformationMetaData.website ?? ''
	);

	Listener({ setFullAddress, setPhoneNumber, setWebsite });

	const getLatLngData = useDebounce(() => {
		console.log('test');
		let lat = 0;
		let lng = 0;
		fetch(`https://nominatim.openstreetmap.org/search?q=${fullAddress}&format=geojson`).then(response => {
				// Check if the response is successful
				if (!response.ok) {
					throw new Error(
						sprintf(__('Network response was not ok %s', 'gatherpress'), response.statusText)
					);
				}
				// Parse the JSON from the response
				return response.json();
			})
			.then((data) => {
				// Process the data
				if (data.features.length > 0) {
					lat = data.features[0].geometry.coordinates[1];
					lng = data.features[0].geometry.coordinates[0];
				}
				updateVenueMeta({
					latitude: lat,
					longitude: lng,
				});
			})
			.catch((error) => {
				// Handle any errors
				console.error(
					sprintf(__('There was a problem with the fetch operation: %s', 'gatherpress'), error)
				);
			});
	}, 2000);

	return (
		<>
			<TextControl
				label={__('Full Address', 'gatherpress')}
				value={fullAddress}
				onChange={(value) => {
					Broadcaster({ setFullAddress: value });
					updateVenueMeta({ fullAddress: value });
					getLatLngData();
				}}
			/>
			<TextControl
				label={__('Phone Number', 'gatherpress')}
				value={phoneNumber}
				onChange={(value) => {
					Broadcaster({ setPhoneNumber: value });
					updateVenueMeta({ setPhoneNumber: value });
				}}
			/>
			<TextControl
				label={__('Website', 'gatherpress')}
				value={website}
				type="url"
				onChange={(value) => {
					Broadcaster({ setWebsite: value });
					updateVenueMeta({ setWebsite: value });
				}}
			/>
		</>
	);
};

export default VenueInformation;
```

Reply:
Just from a first look and probably unrelated to your last question:

- You are right, `useDebounce` is at home at the compose package. Sorry for that!
- Is the `sprintf` function imported (or declared) anywhere?

Reply:
Great catch... do you know if wordpress has that function hosted in any of their packages? Tried checking the docs but didn't get far

Reply:
Oops, sorry I missed that. 

Yes, it's contained in the i18n package like `__` or `_x`.

Reply:
Not yet. Because of the *Map not found error*, I wasn’t able to load any OSM at all. But now, that you fixed that, I’ll be happy to have a look again today.